src/What4/ExprHelpers.hs

```diff
@@ -108,6 +109,13 @@ import qualified What4.Symbol as WS
 import           Data.Parameterized.SetF (SetF)
 import qualified Data.Parameterized.SetF as SetF
 
+
+-- FIXME: this is a workaround for the fact that 'natToInteger' from What4.Interface
+-- is in the IO monad, but doesn't need to be.
+natToIntegerPure ::
+  W4.IsExprBuilder sym => sym -> W4.SymNat sym -> W4.SymInteger sym
+natToIntegerPure _sym n = unsafeCoerce n
```

Comment:
I was hoping for a change in the What4 interface, since this function really does not need to be in IO. I raised an issue here: https://github.com/GaloisInc/what4/issues/193.

Plumbing through the necessary IO to get this working is non-trivial, since this is at the bottom of basically anything that looks at ground values. 

An alternative fix is to address this issue here: https://github.com/GaloisInc/pate/issues/195 by making `integerToNat` more stable (such that unwrapping and wrapping a SymNat does not change the inner term at all).

Reply:
I think the necessary What4 change should just require a few lines to export a new, pure, extraction function.  I don't see any problem with widening the API this way.  We already have a What4 submodule, so no need to wait for a release.

Reply:
Thanks. I just wanted to make sure there wasn't some future API change you had in mind that would be in conflict